chirho/explainable/handlers/components.py

```diff
@@ -292,16 +278,13 @@ def _consequent_eq_neq(consequent: T) -> torch.Tensor:
 
         FACTUAL_NEC_SUFF = torch.zeros_like(sufficiency_log_probs)
 
+        index_keys = set(antecedents)
+        null_index = IndexSet(**{name: {0} for name in index_keys})
+
         nec_suff_log_probs_partitioned = {
+            **{null_index: FACTUAL_NEC_SUFF},
             **{
-                factual_indices: FACTUAL_NEC_SUFF,
-            },
-            **{
-                IndexSet(**{antecedent: {ind}}): log_prob
-                for antecedent in (
-                    set(antecedents)
-                    & set(indices_of(consequent, event_dim=support.event_dim))
```

Comment:
Was this line restricting the index variables to just those already present in `consequent` the root cause of a problem? If so, which of the new or updated tests in this PR fail with this condition in place and pass with the changes?